### PR TITLE
Jonmv/serialize job type with zone

### PR DESCRIPTION
**Pull request by jonmv**
@freva please review. Must be merged _after_ https://github.com/vespa-engine/vespa/pull/22100 has rolled out.

### CODE REVIEW

File: hosted-tenant-base/pom.xml [hunk 1]
```diff
@@ -32,7 +32,7 @@
     </scm>
 
     <properties>
-        <vespaversion>${project.version}</vespaversion>
+        <vespaversion>7-SNAPSHOT</vespaversion>
```
Comment (freva) [inline hosted-tenant-base/pom.xml:35]: Intentional?

**Review by freva — APPROVED** [review]

File: hosted-tenant-base/pom.xml (hunk 1 above)
Comment (jonmv) [reply hosted-tenant-base/pom.xml:35]: Oops, no.

**Comment by jonmv** [comment]
Ah, there were more places jobs were serialised, which I need to fix in both the previous PR and this ...

**Comment by jonmv** [comment]
This PR now switches to serialisation with zone everywhere, and `JobType.fromJobName` is invoked with zones elsewhere. The last PR will remove the compatibility deserialisation code, and let all `JobType` objects hold a specific zone. `systemTest` and `stagingTest` will depend on the zone registry. Most uses are in unit tests, where the `ZoneRegistryMock` can now have a static set of zones, and then also a static member for those two job types; elsewhere, the controller or job controller can be used to instantiate them, or just check the environment of the job zones, if the current usage is `xxx.equals(systemTest)` .

**Review by freva — APPROVED** [review]

**Comment by jonmv** [comment]
What, more conflicts?

**Comment by jonmv** [comment]
:D

**Review by freva — APPROVED** [review]

**Pull request merged**